pkg/compact/compact.go

```diff
@@ -950,6 +955,17 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 			return errors.Wrap(err, "build compaction groups")
 		}
 
+		ignoreDirs := []string{}
+		for _, gr := range groups {
+			ignoreDirs = append(ignoreDirs, gr.Key())
+		}
+
+		files, _ := ioutil.ReadDir(c.compactDir)
+		level.Warn(c.logger).Log("files", fmt.Sprintf("%+v", files), "ignoringdirs", fmt.Sprintf("%+v", ignoreDirs))
```

Comment:
Let's add `msg` to tell what this is warning about (:

Reply:
This was good, old printf-debugging in e2e tests that I forgot to remove and hence not needed since I have figured out the problem since. The culprit was that I had forgotten to remove the `replica` label before trying to get the grouping key for the tests, and then the directory that I had been expecting was removed.